Jint.Benchmark/ShadowRealmBenchmark.cs

```diff
@@ -0,0 +1,61 @@
+﻿using BenchmarkDotNet.Attributes;
+using Esprima.Ast;
+
+namespace Jint.Benchmark;
+
+[MemoryDiagnoser]
+[BenchmarkCategory("ShadowRealm")]
+public class ShadowRealmBenchmark
+{
+    private const string sourceCode = @"
+(function (){return 'some string'})();
+";
+
+    private Engine engine;
+    private Script parsedScript;
+
+    [GlobalSetup]
+    public void Setup()
+    {
+        engine = new Engine();
+        parsedScript = new Esprima.JavaScriptParser().ParseScript(sourceCode);
```

Comment:
you should use `Engine.PrepareScript(sourceCode)`

Reply:
aah... interesting.
Is this only for performance tests, or should we also use `Engine.PrepareScript(xxx)` instead of `new Esprima.JavaScriptParser().ParseScript(sourceCode);` in our production code?

Reply:
done - does not seem to make too much of a difference though:
|                         Method |        Mean |     Error |    StdDev |   Gen0 |   Gen1 | Allocated |
|------------------------------- |------------:|----------:|----------:|-------:|-------:|----------:|
|                  ReusingEngine |  3,921.6 ns |  77.99 ns | 152.12 ns | 0.5417 |      - |   3.34 KB |
|              NewEngineInstance | 10,940.6 ns | 216.49 ns | 361.71 ns | 2.7618 | 0.1373 |  16.93 KB |
|                    ShadowRealm |  9,524.8 ns | 189.78 ns | 454.70 ns | 2.2888 | 0.1526 |  14.06 KB |
|     ReusingEngine_ParsedScript |    911.0 ns |  17.50 ns |  15.52 ns | 0.3204 | 0.0019 |   1.97 KB |
| NewEngineInstance_ParsedScript |  5,345.5 ns |  77.00 ns |  94.57 ns | 2.5253 | 0.1297 |  15.51 KB |
|       ShadowRealm_ParsedScript |  4,677.5 ns |  71.50 ns |  63.38 ns | 2.0599 | 0.1373 |  12.64 KB |

Reply:
You should always prefer `Engine.PrepareScript()` as it's the path that will get optimizations if they are possible. When script goes through the helper we can be assured that we can take our time to analyze more try to pre-calculate things. Otherwise Jint expects that this script might not be run multiple times (it doesn't know).

I recently pushed engine construction optimizations to main so you might want to rebase against it and see if things are better. `NewEngineInstance` cases should be faster at least.